scipy/ndimage/tests/test_filters.py

```diff
@@ -869,6 +869,21 @@ def test_gaussian_gradient_magnitude02(self, dtype):
         expected = tmp1 * tmp1 + tmp2 * tmp2
         expected = numpy.sqrt(expected).astype(dtype)
         assert_array_almost_equal(expected, output)
+    
+    @pytest.mark.parametrize('dtype',
+                             [numpy.int32, numpy.float32, numpy.float64,
+                              numpy.complex64, numpy.complex128])
+    def test_gaussian_gradient_magnitude03(self, dtype):
+        array = numpy.array([[3, 2, 5, 1, 4],
+                             [5, 8, 3, 7, 1],
+                             [5, 6, 9, 3, 5]], dtype) * 100
+        tmp1 = ndimage.gaussian_filter(array, 1.0, [1, 0], mode="reflect-odd")
+        tmp2 = ndimage.gaussian_filter(array, 1.0, [0, 1], mode="reflect-odd")
+        output = numpy.zeros(array.shape, dtype)
+        ndimage.gaussian_gradient_magnitude(array, 1.0, output, mode="reflect-odd")
+        expected = tmp1 * tmp1 + tmp2 * tmp2
+        expected = numpy.sqrt(expected).astype(dtype)
+        assert_array_almost_equal(expected, output)
 
```

Comment:
I would suggest also adding another test case that verifies the extension behavior matches what is obtained via explicitly padding with `numpy.pad` and then cropping the output.

I verified locally that the following test passes

```Python

    def test_uniform_reflect_odd(self):
        array = np.arange(16, dtype=float).reshape((4, 4))

        size = 3
        npad = size//2
        output = ndimage.uniform_filter(array, size, mode='reflect-odd')

        prepadded = np.pad(array, npad, mode='reflect', reflect_type='odd')
        expected = ndimage.uniform_filter(
            prepadded, size, mode='constant')[npad:-npad, npad:-npad]
        assert_allclose(output, expected)
```

Reply:
On closer look, I think this test case should pass on all odd sizes, but currently it fails on some. For example I see failures for size=5 or size=9, but it passes for size=3 and size=7